rpm/src/main/java/org/eclipse/packager/rpm/signature/PgpHeaderSignatureProcessor.java

```diff
@@ -105,6 +110,24 @@ public void feedPayloadData(final ByteBuffer data) {
 
     @Override
     public void finish(final Header<RpmSignatureTag> signature) {
-        signature.putBlob(RpmSignatureTag.RSAHEADER, this.value);
+        switch (this.signature.getKeyAlgorithm()) {
+        // RSA
+        case PublicKeyAlgorithmTags.RSA_GENERAL:
+        case PublicKeyAlgorithmTags.RSA_ENCRYPT:
+        case PublicKeyAlgorithmTags.RSA_SIGN:
+            signature.putBlob(RpmSignatureTag.RSAHEADER, this.value);
```

Comment:
Looks like the only constant we want here is `RSA_GENERAL = `.

https://github.com/rpm-software-management/rpm/blob/0e9cff51e43703ebe248e4d098dad88ca2103f15/sign/rpmgensig.c#L156-L169

Reply:
RSA_SIGN has been deprecated for quite a while (I think it was used by v3 or v2 keys only?).
But since we are consuming here, I though catching the superfluous algorithm tags wouldn't hurt.

I'll remove them though :)

Reply:
I am just going by what the RPM C code seems to support. As you said, there's really no good reason we couldn't support any algorithm that PGP supports, but I think at least RPM would need to add it's own constant for the RPM header first.